### ISSUE TITLE
custom ElasticsearchRepository  "deleteBy<SomeField>In (Iterable<String> values)" deletes ALL documents.  

### ISSUE THREAD

**Issue by mancini0** [VanRoy/spring-data-jest#58]
1) Create an ElasticsearchRepository interface with a custom "deleteBy...In" method

```
public interface ProductRepository extends ElasticsearchRepository<Product,String> {
    public Long deleteByProductGuidIn(Iterable<String> productGuids);
}
```

2) The associated Product `@Document `annotated class looks as follows:

Some notes:

2.a) In this case, the field I am deleting by is an `@Id` field, but I have observed this behavior with non-Id annotated fields. 

2.b)I have not tested the behavior on non `@Field(type = FieldType.keyword)` annotated fields, the fields I am interested in deleting by are all of type keyword.

```
@Getter
@Setter
@Document(indexName = "products", type = "product")
public class Product implements Serializable {
    @Id
    @Field(type = FieldType.keyword)
    private String productGuid;
    @Field(type = FieldType.keyword)
    private String securityType;
    @Field(type = FieldType.keyword)
    private String clearingSymbol;
    
   @Field(type = FieldType.keyword)
    private String exchangeClearing;
    @Field(type = FieldType.keyword)
    private String exchangeGlobex;
... other fields omitted ...
}
```

3) Persist a few Product documents to Elasticsearch

@Autowired
ProductRepository productRepo
 ...
Product p1 = new Product();
p1.setProductGuid("abc123");
p1.setSecurityType("FUT");

Product p2 = new Product();
p2.setProductGuid("xyz456");
p2.setSecurityType("OPT");

Product p3 = new Product();
p2.setProductGuid("ghi987");
p2.setSecurityType("CDS");

List<Product> products = new ArrayList<>();
products.add(p1); 
products.add(p2);
products.add(p3);

productRepo.saveAll(products);
...

4)Call the custom deleteBy...In() method:
List<String> idsToDelete = new ArrayList<>();
idsToDelete.add("abc123");
idsToDelete.add("xyz456");
productRepo.deleteByProductGuidIn(idsToDelete) 

5)Observe that all documents have been removed from elasticsearch, not just the ones having an id specified in step 4.

**Comment by VanRoy** [comment]
Hi @mancini0 , thanks for your feedback. Can you tell me which version of Spring Data Jest do you use ? 
I will investigate on it.

Julien.

**Comment by VanRoy** [comment]
Hi @mancini0 , your issue is related to this : https://github.com/VanRoy/spring-data-jest/pull/59

It fixed in 3.1.2.RELEASE released today.
